src/doc/reference.md

```diff
@@ -2959,10 +2956,10 @@ meaning of the operators on standard types is given here.
   : Exclusive or.
     Calls the `bitxor` method of the `std::ops::BitXor` trait.
 * `<<`
-  : Logical left shift.
+  : Aritmetic left shift.
```

Comment:
Executive summary: considering how short this text already is, you should not say "Logical" nor "Arithmetic".  Just say "Left shift" and "Right shift."

----

Further explanation:

First off, typo: "Arithmetic", not "Aritmetic"

But more importantly: The "Arithmetic" versus "Logical" distinction does not arise for the left shift operator `Shl` (`<<`): it will always just blindly shift the bits over (and overwrite the sign bit), and shift in zeroes into the least significant bits.

The place where the Arithmetic versus Logical comes up is for *right-shifts*, since it determines what to do when you have a non-zero sign bit that you are shifting to the right.  Namely: Should one shift in *zeroes* into the most significant bits in such a case, or should one shift in *ones* ?
 * When one is looking at the value as a finite sequence of bits, then it often seems intuitive to shift in zero.
 * But when one is looking at the value under an arbitrary-precision twos-complement interpretation of the bits, the sign bit of 1 actually represents an leading infinite sequence of ones.

See also http://en.wikipedia.org/wiki/Arithmetic_shift

----

The way that we handle this issue in Rust: The `Shr` operator itself does not dictate whether it performs Logical or Arithmetic shift; that choice is up to the implementer of the trait.

So for the builtin numerics is that we decide based on the type of the left-hand side: For an unsigned integral type, right-shift is logical.  For a signed integral type, right-shift is arithmetic.  (So we support both logical and arithmetic shift, without adding distinct operators for them the way that e.g. Java does with its `>>` versus `>>>` distinction.)